test/integration/linux-6.2.0/module.patch

```diff
@@ -0,0 +1,95 @@
+From 4ad0397efd708a961ecf733bc649783565a6f2db Mon Sep 17 00:00:00 2001
+From: Joe Lawrence <joe.lawrence@redhat.com>
+Date: Thu, 19 Jan 2023 16:43:46 -0500
+Subject: [PATCH] kpatch module integration test
+Content-type: text/plain
+
+This tests several things related to the patching of modules:
+
+- 'kpatch_string' tests the referencing of a symbol which is outside the
+  .o, but inside the patch module.
+
+- alternatives patching (.altinstructions)
+
+- paravirt patching (.parainstructions)
+
+- jump labels (5.8+ kernels only) -- including dynamic printk
+
+Signed-off-by: Josh Poimboeuf <jpoimboe@redhat.com>
+---
+ fs/nfsd/export.c         | 31 +++++++++++++++++++++++++++++++
+ net/netlink/af_netlink.c |  5 +++++
+ 2 files changed, 36 insertions(+)
+
+diff --git a/fs/nfsd/export.c b/fs/nfsd/export.c
+index 668c7527b17e..a3122ee61466 100644
+--- a/fs/nfsd/export.c
++++ b/fs/nfsd/export.c
+@@ -1294,6 +1294,27 @@ static void exp_flags(struct seq_file *m, int flag, int fsid,
+ 	}
+ }
+ 
++#include "kpatch-macros.h"
++static int pre_patch_callback(patch_object *obj)
++{
++#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
++	static DEFINE_STATIC_KEY_TRUE(kpatch_key);
++
++	if (static_branch_unlikely(&memcg_kmem_enabled_key))
++		printk("kpatch: memcg_kmem_enabled_key\n");
++
++	BUG_ON(!static_branch_likely(&kpatch_key));
++	static_branch_disable(&kpatch_key);
++	BUG_ON(static_branch_likely(&kpatch_key));
++	static_branch_enable(&kpatch_key);
++#endif
++       return 0;
++}
++KPATCH_PRE_PATCH_CALLBACK(pre_patch_callback);
```

Comment:
@jpoimboe : what do you think of this change?  I don't know why we've never seen dmesg complaints before, but static_branch_disable() may block and provoke RCU read-side critical section warnings.  Moving this to a call back puts it in a safe context, but I didn't know if there were other reasons for these changes to exist in a replaced function (and not a callback).

Reply:
Hm, I think moving it to a callback defeats the point of the test.  It's intended to test jump labels in patched code, which rely on create-diff-object and klp relas working correctly.